sdk/go/x/auto/example_test.go

```diff
@@ -205,6 +206,63 @@ func ExampleGitRepo() {
 	NewStackRemoteSource(ctx, fqsn, repo, Project(project))
 }
 
+func ExampleGitRepo_personalAccessToken() {
+	ctx := context.Background()
+	pName := "go_remote_proj"
+	fqsn := FullyQualifiedStackName("myOrg", pName, "myStack")
+
+	// Get the Sourcecode Repository PERSONAL_ACCESS_TOKEN
+	token, _ := os.LookupEnv("PERSONAL_ACCESS_TOKEN")
+
+	repo := GitRepo{
+		URL:         "https://github.com/pulumi/test-repo.git",
+		ProjectPath: "goproj",
+		Auth: &GitAuth{
+			PersonalAccessToken: token,
+		},
+	}
+
+	// initialize a stack from the git repo, specifying our project override
+	NewStackRemoteSource(ctx, fqsn, repo)
+}
+
+func ExampleGitRepo_privateKeyPath() {
+	ctx := context.Background()
+	pName := "go_remote_proj"
+	fqsn := FullyQualifiedStackName("myOrg", pName, "myStack")
+
+	repo := GitRepo{
+		URL:         "https://github.com/pulumi/test-repo.git",
```

Comment:
Interesting, I thought you'd have to specify the URL like `git@github.com:pulumi/test-repo.git` when using SSH for auth, but it works when doing HTTPS?. Do the `git@...` style URLs work as well?

Reply:
they do AFAICT

Reply:
@stack72 Just ran into this for the operator, and the URL does in fact need to be `git@github.com...` as Justin points out.

Using the HTTPS url results in `unable to clone repo: invalid auth method`

Reply:
Opened https://github.com/pulumi/pulumi/issues/5382